Zend/zend_inheritance.c

```diff
@@ -672,9 +676,43 @@ static inheritance_status zend_is_intersection_subtype_of_type(
 }
 
 ZEND_API inheritance_status zend_perform_covariant_type_check(
-		zend_class_entry *fe_scope, const zend_type fe_type,
-		zend_class_entry *proto_scope, const zend_type proto_type)
+		zend_class_entry *fe_scope, const zend_type *fe_type_ptr,
+		zend_class_entry *proto_scope, const zend_type *proto_type_ptr);
+
+static inheritance_status zend_is_type_subtype_of_generic_type(
+	zend_class_entry *concrete_scope,
+	const zend_type *concrete_type_ptr,
+	zend_class_entry *generic_type_scope,
+	const zend_type *generic_type_ptr
+) {
+	ZEND_ASSERT(concrete_scope->bound_types);
```

Comment:
This breaks when overriding interface method:
```php
interface I<T> {
    function f(T $t);
}

interface J<T> extends I<T> {
    function f(T $t);
}
```

Reply:
Fixed with the latest pushes, thank you for the test case!